src/Npgsql/NpgsqlCommand.cs

```diff
@@ -480,8 +480,8 @@ void DeriveParametersForQuery(NpgsqlConnector connector)
 
                 foreach (var statement in _statements)
                 {
-                    Expect<ParseCompleteMessage>(connector.ReadMessage(), connector);
-                    var paramTypeOIDs = Expect<ParameterDescriptionMessage>(connector.ReadMessage(), connector).TypeOIDs;
+                    Expect<ParseCompleteMessage>(connector.ReadMessageWithoutCancellation(), connector);
```

Comment:
Any reason this shouldn't do PG cancellation?

Reply:
This one is... actually, I'm not really sure about. Does it really can timeout because it was executing for too long?

Reply:
Probably not - we're just asking the server to parse and describe the parameters. I think in my mind I wanted to default to trying cancellation, but like with COPY we can go with this and revisit later (this API specifically is in very low usage.